packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1091,6 +1105,28 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
 
       final double xOffset = hasCustomWidth ? (size.width - snackBarWidth!) / 2 : 0.0;
       positionChild(_ScaffoldSlot.snackBar, Offset(xOffset, snackBarYOffsetBase - snackBarSize.height));
+
+      assert((){
+        // Whether a SnackBar has been offsetted too high by floatingActionButton.
+        // Assert after call to positionChild otherwise SnackBar position differs in debug mode.
+        if (floatingActionButtonRect.size != Size.zero && isSnackBarFloating) {
```

Comment:
This `if` statement already exists above on line 1092, can this just be included in that block?